native-schema-registry/src/main/java/com/amazonaws/services/schemaregistry/serializer/ProtobufPreprocessor.java

```diff
@@ -0,0 +1,39 @@
+package com.amazonaws.services.schemaregistry.serializer;
+
+import com.amazonaws.services.schemaregistry.deserializers.protobuf.ProtobufSchemaParser;
+import com.amazonaws.services.schemaregistry.serializers.protobuf.MessageIndexFinder;
+import com.amazonaws.services.schemaregistry.serializers.protobuf.ProtobufWireFormatEncoder;
+import com.amazonaws.services.schemaregistry.utils.ProtobufUtils;
+import com.google.common.collect.BiMap;
+import com.google.protobuf.DescriptorProtos;
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.InvalidProtocolBufferException;
+
+import java.io.IOException;
+import java.util.Base64;
+
+public class ProtobufPreprocessor {
+    private static final ProtobufWireFormatEncoder ENCODER = new ProtobufWireFormatEncoder(new MessageIndexFinder());
+    private static final MessageIndexFinder MESSAGE_INDEX_FINDER = new MessageIndexFinder();
+
+    public static String convertBase64SchemaToStringSchema(String base64Schema) throws InvalidProtocolBufferException {
+        DescriptorProtos.FileDescriptorProto
+            fileDescriptorProto =
+            DescriptorProtos.FileDescriptorProto.parseFrom(Base64.getDecoder().decode(base64Schema));
+
+        return ProtobufUtils.getProtobufSchemaStringFromFileDescriptorProto(fileDescriptorProto);
+    }
+
+    public static byte[] prefixMessageIndexToBytes(byte[] bytesToEncode, String schemaDef, String descriptorFullname)
+        throws Descriptors.DescriptorValidationException, IOException, NoSuchMethodException {
+        Descriptors.FileDescriptor fileDescriptor = ProtobufSchemaParser.parse(schemaDef, "dummy.proto");
```

Comment:
What's the significance of `dummy.proto`? If it is not being used, you can name it something like `any-name.proto`

Reply:
renamed